lib/api_client.dart

```diff
@@ -51,9 +51,9 @@ class ApiClient {
   // If collectionFormat is 'multi' a key might appear multiple times.
   Future<ApiResponse> invokeAPI(String path, Iterable<QueryParam> queryParams,
       Object body, List<String> authNames, Options options) async {
-    _updateParamsForAuth(authNames, queryParams, options.headers);
+    _updateParamsForAuth(authNames, queryParams as List<QueryParam>, options.headers);
```

Comment:
yeah, this isn't any good - its actually very bad code. It should create a new list, add the existing query params into it and then pass it in. We can't rely on the queryParams being a List and being mutable.